src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -7,159 +7,150 @@ import akka.http.scaladsl.server.Directives._
-      (headerValueByName(authorizationHeader) &
-        provide(identityConcentratorService)) tflatMap {
-        // order of these cases is important: if the authorization header is a valid jwt we must ignore
-        // any OIDC headers otherwise we may be vulnerable to a malicious user specifying a valid JWT
-        // but different user information in the OIDC headers
-        case (JwtAuthorizationHeader(Success(jwtUserInfo), bearerToken), Some(icService)) =>
-          withChildTraceSpan("JWT-request", samRequestContext).flatMap { samRequestContext =>
-            fromJwt(jwtUserInfo, bearerToken, icService, samRequestContext)
-          }
-
-        case (JwtAuthorizationHeader(Failure(regrets), _), _) =>
-          failWith(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Unauthorized, "could not parse authorization header, jwt missing required fields", regrets)))
-
-        case (JwtAuthorizationHeader(_, _), None) =>
-          logger.error("requireUserInfo with JWT attempted but Identity Concentrator not configured")
-          failWith(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, "Identity Concentrator not configured")))
-
-        case _ =>
-          // request coming through Apache proxy with OIDC headers
-          withChildTraceSpan("OIDC-request", samRequestContext).flatMap { samRequestContext =>
-            (headerValueByName(accessTokenHeader).as(OAuth2BearerToken) &
-              headerValueByName(googleSubjectIdHeader).as(GoogleSubjectId) &
-              headerValueByName(expiresInHeader) &
-              headerValueByName(emailHeader).as(WorkbenchEmail)).as(OIDCHeaders).flatMap(fromOIDC(_, samRequestContext))
-          }
-      }
-    }
-
-  private def requireUserInfoFromJwt(jwtUserInfo: JwtUserInfo, bearerToken: OAuth2BearerToken, icService: IdentityConcentratorService, samRequestContext: SamRequestContext): Directive1[UserInfo] =
+  private def populateGoogleIdAndThrowIfUserExists(samRequestContext: SamRequestContext, accessToken: OAuth2BearerToken, googleOpaqueTokenResolver: GoogleOpaqueTokenResolver, azureB2CId: AzureB2CId, createWorkbenchUser: CreateWorkbenchUser): Directive1[CreateWorkbenchUser] = {
     onSuccess {
-      getUserInfoFromJwt(jwtUserInfo, bearerToken, directoryDAO, icService, samRequestContext).unsafeToFuture()
+      googleOpaqueTokenResolver.getWorkbenchUser(accessToken, samRequestContext).flatMap {
+        case None =>
+          IO.pure(createWorkbenchUser) // access token not valid
+        case Some(GoogleTokenInfo(None, googleSubjectId)) => // access token valid but not already a user
+          IO.pure(createWorkbenchUser.copy(googleSubjectId = Option(googleSubjectId)))
+        case Some(GoogleTokenInfo(Some(_), _)) => // already a user
+          IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"user $azureB2CId already exists")))
```

Comment:
Perhaps modify the pattern match to capture the `googleSubjectId` and then:

```suggestion
          IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"user with Google Subject ID $googleSubjectId already exists")))
```
Because if I'm reading this right, we're looking up the user record based on their `googleSubjectId`, not their `azureB2CId`.  And I think it's important that in any of these log statements now, we be explicit as to which user ID we are printing.  Yeah, we should be able to tell by if it's a `long` or a `uuid`, but I think it is better to explicit.

Reply:
how do you feel about printing the email address instead? I don't think the id is particularly helpful to the user anyway and neither is the subtly of google or azure id